### PR TITLE
Tooltips for group columns.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/shared/table/table.dart [hunk 1]
```diff
@@ -1521,7 +1521,9 @@ class _TableRowState<T> extends State<TableRow<T>>
           switch (displayTypeForIndex) {
             case _TableRowPartDisplayType.column:
               final current = columnIndexTracker;
-              columnIndexTracker++;
+              if (columnIndexTracker + 1 < widget.columns.length) {
+                columnIndexTracker++;
+              }
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/table/table.dart:1526]: why would we ever have more items in `rowDisplayParts` with type `_TableRowPartDisplayType.column` than we have columns? that seems like a bug with how `rowDisplayParts` is being generated

File: packages/devtools_app/lib/src/shared/table/table_data.dart [hunk 2]
```diff
@@ -113,12 +113,14 @@ mixin PinnableListEntry {
 /// will be drawn between groups and an additional header row will be added to
 /// the table to display the column group titles.
 class ColumnGroup {
-  ColumnGroup({required this.title, required this.range});
+  ColumnGroup({this.tooltip, required this.title, required this.range});
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/table/table_data.dart:116]: nit: put non-required item at the end of the parameter list

File: packages/devtools_app/lib/src/shared/table/table.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/shared/table/table.dart:1526]: it seems to be side effect of scroll overflow

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/table/table.dart:1526]: I dug into this a bit. Fix for this bug is here: https://github.com/flutter/devtools/pull/4606

**Review by kenzieschmoll — APPROVED** [review]
you'll have to merge and resolve the conflicts (i.e. remove this PR's changes to columnIndexTracker), but LGTM after that

**Pull request merged**